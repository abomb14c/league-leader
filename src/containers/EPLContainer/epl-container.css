.epl-header {
  background-color: #3B3738;
  border-bottom: 2px solid #C63D0F;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  display: flex;
  color: #FDF3E7;
  font-weight: 200;
  padding-bottom: 5px;
}

.header-desc {
  font-weight: 600;
  margin-left: 2%;
}

.epl-container {
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  display: flex;
  flex-direction: column;
  height: 900px;
  margin-left: 5%;
  margin-top: 3%;
  width: 45%;
}

.points {
  margin-left: 10%;
}

.record {
  margin-left: 10%;
}

.rank {
  margin-left: 42%;
}