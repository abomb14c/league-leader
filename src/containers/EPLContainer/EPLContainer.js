import React from 'react';
import { connect } from 'react-redux'; 
import EnglishCard from '../../components/EnglishCard/EnglishCard';
import './epl-container.css';
import PropTypes from 'prop-types';

export const EPLContainer = (props) => {
  const eplCards = props.EPL.map((team, index) => {
    return (
      <EnglishCard {...team} key={index} />
    );
  });
  
  return (
    <div className="epl-container">
      <div className="epl-header">
        <h3 className="header-desc">Name</h3>
        <h3 className="header-desc rank">Rank</h3>
        <h3 className="header-desc record">Record</h3>
        <h3 className="header-desc points">Points</h3>
      </div>
      {eplCards}
    </div>
  );
};


export const mapStateToProps = state => ({
  EPL: state.EPL
});

export default connect(mapStateToProps, null)(EPLContainer);

EPLContainer.propTypes = {
  EPL: PropTypes.arrayOf(PropTypes.object)
};