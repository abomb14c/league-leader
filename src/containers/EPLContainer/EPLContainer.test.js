import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import  { EPLContainer, mapStateToProps}  from './EPLContainer';

describe('DraftCard', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {

    mockProps = {
      EPL: [ {}, {}, {}]
    };

    wrapper = shallow(<EPLContainer
      {...mockProps}
    />);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

// describe('mapStateToProps', () => {
//   it('should map the user to props', () => {
//     const mockState = {
//       league: {},
//       EPL: [{}, {}, {}],
//       draftTeams: [{}, {}, {}],
//       text: ''
//     };

//     const expected = {
//       league: {},
//       EPL: [{}, {}, {}],
//       draftTeams: [{}, {}, {}]
//     };
    
//     const mappedProps = mapStateToProps(mockState);
//     expect(mappedProps).toEqual(expected);
//   });
// });
