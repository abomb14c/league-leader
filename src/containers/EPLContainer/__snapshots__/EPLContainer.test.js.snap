// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EPLContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    EPL={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="epl-header"
>
          <h3
                    className="header-desc"
          >
                    Name
          </h3>
          <h3
                    className="header-desc rank"
          >
                    Rank
          </h3>
          <h3
                    className="header-desc record"
          >
                    Record
          </h3>
          <h3
                    className="header-desc points"
          >
                    Points
          </h3>
</div>,
        Array [
          <Unknown />,
          <Unknown />,
          <Unknown />,
        ],
      ],
      "className": "epl-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="header-desc"
>
              Name
</h3>,
            <h3
              className="header-desc rank"
>
              Rank
</h3>,
            <h3
              className="header-desc record"
>
              Record
</h3>,
            <h3
              className="header-desc points"
>
              Points
</h3>,
          ],
          "className": "epl-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name",
              "className": "header-desc",
            },
            "ref": null,
            "rendered": "Name",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rank",
              "className": "header-desc rank",
            },
            "ref": null,
            "rendered": "Rank",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Record",
              "className": "header-desc record",
            },
            "ref": null,
            "rendered": "Record",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Points",
              "className": "header-desc points",
            },
            "ref": null,
            "rendered": "Points",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="epl-header"
>
            <h3
                        className="header-desc"
            >
                        Name
            </h3>
            <h3
                        className="header-desc rank"
            >
                        Rank
            </h3>
            <h3
                        className="header-desc record"
            >
                        Record
            </h3>
            <h3
                        className="header-desc points"
            >
                        Points
            </h3>
</div>,
          Array [
            <Unknown />,
            <Unknown />,
            <Unknown />,
          ],
        ],
        "className": "epl-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="header-desc"
>
                Name
</h3>,
              <h3
                className="header-desc rank"
>
                Rank
</h3>,
              <h3
                className="header-desc record"
>
                Record
</h3>,
              <h3
                className="header-desc points"
>
                Points
</h3>,
            ],
            "className": "epl-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
                "className": "header-desc",
              },
              "ref": null,
              "rendered": "Name",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rank",
                "className": "header-desc rank",
              },
              "ref": null,
              "rendered": "Rank",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Record",
                "className": "header-desc record",
              },
              "ref": null,
              "rendered": "Record",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Points",
                "className": "header-desc points",
              },
              "ref": null,
              "rendered": "Points",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
